src/App.vue

```diff
@@ -286,38 +365,54 @@ function updateLocalHistory() {
     query: { ...unref(route).query, ...query }
   })
 }
+
 async function next() {
   if (!unref(isModelReady)) {
     return
   }
+
   if (unref(activeIndex) + 1 >= unref(modelFiles).length) {
     activeIndex.value = 0
   } else {
     activeIndex.value++
   }
 
   updateLocalHistory()
+  await unloadModels()
   // TODO: how to prevent activeFiles from being reduced
   // load activeFiles
   await loadFolderForFileContext(unref(currentFileContext))
   await renderNewModel()
 }
+
 async function prev() {
   if (!unref(isModelReady)) {
     return
   }
+
   if (unref(activeIndex) === 0) {
     activeIndex.value = unref(modelFiles).length - 1
   } else {
     activeIndex.value--
   }
 
   updateLocalHistory()
+  await unloadModels()
   // TODO: how to prevent activeFiles from being reduced
   // load activeFiles
   await loadFolderForFileContext(unref(currentFileContext))
   await renderNewModel()
 }
+
+async function unloadModels(): Promise<void> {
```

Comment:
could you also check how we can include this as well? or may not be required with this new method
https://github.com/saw-jan/web-app-3dmodel-viewer/blob/bb668d1586ebbb904af5a744650d939a1e06c4a0/src/App.vue#L241

you are right, the cleanup stuff should have it's own method

Reply:
Removed scene.remove from renderNewModel. no longer needed.